generators/python/src/fern_python/snippet/snippet_template_factory.py

```diff
@@ -641,6 +630,10 @@ def get_type_reference_template_input(
         name_breadcrumbs: Optional[List[str]],
         indentation_level: int = 0,
     ) -> Union[TemplateInput, None]:
+        # Terminate if depth is too deep
+        if indentation_level >= self.MAXIMUM_TEMPLATE_DEPTH:
```

Comment:
This is the real change, it uses indentation level to indicate depth, but happy to introduce an explicit depth param